contracts/core/UniversalLiquidator.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+// imported contracts and libraries
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+// interfaces
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../interface/IUniversalLiquidator.sol";
+import "../interface/IUniversalLiquidatorRegistry.sol";
+import "../interface/ILiquidityDex.sol";
+
+// libraries
+import "../libraries/DataTypes.sol";
+import "../libraries/Errors.sol";
+
+contract UniversalLiquidator is Ownable, IUniversalLiquidator {
+    using SafeERC20 for IERC20;
+
+    address public pathRegistry;
+
+    function swapTokenOnMultipleDEXes(

```

Comment:
We might consider a different name for this function, as it is not necessarily about swapping on multiple dexes.

Reply:
What about we make it "swap"?

Reply:
Yeah `swap` works